modules/local/cluster_file/main.nf

```diff
@@ -0,0 +1,44 @@
+process CLUSTER_FILE {
+    tag "Create cluster file for GAS call"
+    label 'process_single'
+
+    input:
+    val meta
+
+    output:
+    path("expected_clusters.txt"), emit: text
+
+    exec:
+    def outputLines = []
+
+    // Determine the maximum number of levels to set the header requirements for each pipeline run
+    int maxLevels = meta.collect { sample -> sample.address.split("\\$params.gm_delimiter").size() }.max() ?: 0
```

Comment:
Since `params.gm_delimiter` may or may not contain special characters like `.`, you can escape the entire string with [Pattern.quote()](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#quote-java.lang.String-). That is:

```groovy
sample.address.split(java.util.regex.Pattern.quote(params.gm_delimiter))
```

I think you have to use the fully qualified name for Pattern here `java.util.regex.Pattern` (or have `import java.util.regex.Pattern` at the top of the file or maybe at the top of the code in the `exec:` block since I'm not sure how Nextflow handles importing packages for `exec:` code blocks).

Reply:
I ended up defining a `delimiter` so it could be called throughout the process. See the change here: [358175e](https://github.com/phac-nml/gasnomenclature/pull/11/commits/358175e565b80330d1f4ce1686dd111b35fe8fad)

Reply:
Added here: [6691dea](https://github.com/phac-nml/gasnomenclature/pull/11/commits/6691dea10c7140403b859fb97ea9a2c4cc79149b)